libr/core/cfile.c

```diff
@@ -907,6 +907,12 @@ R_API RCoreFile *r_core_file_open(RCore *r, const char *file, int flags, ut64 lo
 			swstep = false;
 		}
 		r_config_set_i (r->config, "dbg.swstep", swstep);
+		// Set the correct debug handle
+		if (fd->plugin && fd->plugin->isdbg) {
+			const char *dh = strndup(file, (strstr(file, "://") - file));
```

Comment:
What happens if theres no :// ?

Reply:
Added a check. Is there a point in printing an error or attempting to set the current plugin's name as the debugger in case the uri didn't contain the debugger backend? Seems unlikely with fd->plugin->isdbg